### DISCUSSION TITLE
Feature Request: Limiting Results based on owner type

### DISCUSSION THREAD

**Discussion by avivkeller** [community/community#116817] [Repositories]
I'd like to propose an enhancement to GitHub's search functionality that I believe could significantly improve the way users find repositories and other items on the platform. The addition of a new search parameter, `owned_by`, could offer a more tailored search experience by allowing users to filter search results based on the owner type of the repositories or items.

## New Search Parameter: `owned_by`

- **Usage:** This parameter would enable users to specify the owner type of items they wish to include in their search results.

**Options:**
- `organization`: Limits search results to items owned by GitHub organizations.
- `user`: Filters search results to items owned by individual GitHub users.
- `pro_user`: Focuses search results on items owned by GitHub Pro users, catering to those looking for more professional or advanced projects.

## Use Cases:

**User-Owned Content:** 
A user interested in finding beginner-friendly HTML projects could use the query `language:HTML owned_by:user "Hello World"`. This would return HTML files containing 'Hello World', specifically owned by individual users, likely to include personal or smaller-scale projects.

**Organization-Owned Content:** 
Conversely, someone looking for more established or collaborative HTML projects could use `language:HTML owned_by:organization "Hello World"`. This search would yield HTML files containing 'Hello World' that are owned by organizations, potentially indicating more structured or community-backed projects.

> [!NOTE]
> Format improved via the comment by @giovannivicentin

**Comment by giovannivicentin** [comment]
Hello, RedYetiDev!

I'd like to propose an enhancement to GitHub's search functionality that I believe could significantly improve the way users find repositories and other items on the platform. The addition of a new search parameter, `owned_by`, could offer a more tailored search experience by allowing users to filter search results based on the owner type of the repositories or items.

**Proposed Feature:**
- **New Search Parameter**: `owned_by`
  - **Usage**: This parameter would enable users to specify the owner type of items they wish to include in their search results.
  - **Options**:
    - `organization`: Limits search results to items owned by GitHub organizations.
    - `user`: Filters search results to items owned by individual GitHub users.
    - *(Optional)* `pro_user`: Focuses search results on items owned by GitHub Pro users, catering to those looking for more professional or advanced projects.

**Examples of Use Cases:**
1. **User-Owned Content**: A user interested in finding beginner-friendly HTML projects could use the query `language:HTML owned_by:user "Hello World"`. This would return HTML files containing 'Hello World', specifically owned by individual users, likely to include personal or smaller-scale projects.

2. **Organization-Owned Content**: Conversely, someone looking for more established or collaborative HTML projects could use `language:HTML owned_by:organization "Hello World"`. This search would yield HTML files containing 'Hello World' that are owned by organizations, potentially indicating more structured or community-backed projects.

This enhancement could streamline the search process for users looking for specific types of projects, whether they are more interested in grassroots, individual endeavors or more formal, organizational contributions. It could also aid in educational or research contexts where the distinction between individual and organizational work is relevant.

I believe this feature could add significant value to GitHub's search capabilities and would be appreciated by many users looking to refine their search criteria further.

Thank you for considering this suggestion. I look forward to any feedback or discussion on this proposal.

**Comment by avivkeller** [comment]


**Reply by giovannivicentin** [reply]
Exactly! If it works, tell me. ❤️

**Comment by souravkrshaw** [comment]
I think it will be a reasonable and useful feature to have. I really appreciate if it gets added
